completion/available/aliases.completion.bash

```diff
@@ -20,28 +20,42 @@ function _bash-it-component-completion-callback-on-init-aliases() {
 	completions=("${completions[@]##complete -* * -}") # strip all but last option plus trigger(s)
 	completions=("${completions[@]#complete -}")       # strip anything missed
 	completions=("${completions[@]#? * }")             # strip last option and arg, leaving only trigger(s)
+	completions=("${completions[@]#? }")       # strip anything missed
+	#TODO: this will fail on some completions...
 
 	# create temporary file for wrapper functions and completions
 	tmp_file="$(mktemp -t "${namespace}-${RANDOM}XXXXXX")" || return 1
 
-	completion_loader="$(complete -p -D 2> /dev/null | sed -Ene 's/.* -F ([^ ]*).*/\1/p')"
+	IFS=$'\n' read -r completion_loader < <(complete -p -D 2> /dev/null)
+	if [[ "${completion_loader#complete }" =~ '-F'[[:space:]]([[:alnum:]_]+)[[:space:]] ]]; then
+		completion_loader="${BASH_REMATCH[1]}"
+	else
+		completion_loader=""
+	fi
 
 	# read in "<alias> '<aliased command>' '<command args>'" lines from defined aliases
 	# some aliases do have backslashes that needs to be interpreted
 	# shellcheck disable=SC2162
 	while read line; do
+		line="${line#alias -- }"
 		line="${line#alias }"
 		alias_name="${line%%=*}"
-		alias_defn="${line#*=}"                 # alias definition
+		alias_defn="${line#*=\'}"                 # alias definition
+		alias_defn="${alias_defn%\'}"
 		alias_cmd="${alias_defn%%[[:space:]]*}" # first word of alias
-		alias_cmd="${alias_cmd:1}"              # lose opening quotation mark
-		alias_args="${alias_defn#*[[:space:]]}" # everything after first word
-		alias_args="${alias_args%\'}"           # lose ending quotation mark
+		if [[ ${alias_defn} == ${alias_cmd} ]]; then
+			alias_args=''
+		else
+			alias_args="${alias_defn#*[[:space:]]}" # everything after first word
+		fi
 
 		# skip aliases to pipes, boolean control structures and other command lists
-		[[ "${alias_args}" =~ [\|\&\;\)\(\n] ]] && continue
+		chars='\|\&\;\)\(\n\<\>'
+		if [[ "${alias_defn}" =~ [$chars] ]]; then
+			continue
+		fi
 		# avoid expanding wildcards
-		read -a alias_arg_words <<< "$alias_args"
+		read -ra alias_arg_words <<< "$alias_args"
```

Comment:
I think we went back and forth with this change, we had a couple of PRs that touched this. Are you sure we want to do this?

Reply:
`-r` should be used in all cases where we want to preserve the input. It should be left out *only* if we want to interpret the input, that is, if we're reading from the terminal and the user may be entering a long line or pasting something. 

We *do* leave it out for the `while` loop reading from `alias -p` to ensure that we get the whole alias definition on one (logical) line, which is why the check is disabled.

It's important not to mangle the alias definitions, so we need `-r`.